sycl/include/sycl/handler.hpp

```diff
@@ -2150,6 +2191,50 @@ class __SYCL_EXPORT handler {
     parallel_for_impl<KernelName>(Range, Properties, std::move(KernelFunc));
   }
 
+  template <typename KernelName, typename KernelType, int Dims,
+            typename PropertiesT>
+  void parallel_for_large_range(range<Dims> NumWorkItems, PropertiesT Props,
+                                KernelType KernelFunc) {
+    auto Dev = detail::getDeviceFromHandler(*this);
+    size_t WGSize = Dev.get_info<info::device::max_work_group_size>();
```

Comment:
Those lines should be moved closer to the first use of those variables. Currently this code always queries `max_work_group_size` device info, but may return error without even using the query result.